smart_contract/contracts/CrowdFunding.sol

```diff
@@ -241,4 +259,64 @@ contract CrowdFunding {
             }
         }
     }
+
+    struct DonationData {
```

Comment:
I think it's better to place this at the top as well🙇
Can this be reused for the return data in `Donations` tab also? If yes, please make changes so that it can be reused (like renaming the `address donator` to something more general)